docs/sources/http_api/query_history.md

```diff
@@ -77,6 +77,62 @@ Status codes:
 - **400** - Errors (invalid JSON, missing or invalid fields)
 - **500** – Unable to add query to the database
 
+## Query history search
+
+`GET /api/query-history`
+
+Returns a list of queries in query history that match search parameters. Query history search supports pagination. Use the `limit` query parameter to control the maximum number of queries returned; the default limit is 100. You can also use the `page` query parameter to fetch queries from any page other than the first one.
+
+Query parameters:
+
+- **datasourceUid** - Use this parameter to filter query history items for selected data source. At least 1 data source uid needs to be specified. To do an "AND" filtering with multiple data sources, specify the data source parameter multiple times e.g. `datasourceUid=uid1&datasourceUid=uid2`.
+- **searchString** – Use searchString parameter for filtering query history items based on the content of queries.
+- **sort** - Sorting order. Can be `time-asc` or `time desc`. Defaults to `time-desc`.
+- **onlyStarred** - Search for queries that are starred. Defaults to `false`.
+- **page** - Number of page you would like to to get. Search supports pagination and it is limits results to 100 queries per page.
```

Comment:
There's also `limit` property that doesn't seem to be documented.

Reply:
Fixed in https://github.com/grafana/grafana/pull/45932/commits/8208e0613f6f0f8151caf5589f84b8099c7a32b7